### ISSUE TITLE
`no-empty-first-line` ends the lint in an error on a stylesheet holding a free semicolon and no node

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#602]
The check of `no-empty-first-line` passes over a stylesheet whose text trims to nothing and reports every other one that opens with a line break; its fix takes that break out of `root.first.raws.before`, and throws where the root has no first node. A stylesheet holding a free semicolon and nothing else is both: `"\n;"` does not trim to nothing, PostCSS files the semicolon into the root's trailing raw and gives the root no node, and `--fix` ends the whole lint in an error rather than in a warning:

```js
if (root.first === undefined) throw new Error(`The root node must have the first node.`)
```

## The runs

On `28fe7bc`, through `stylelint.lint` with `fix: true` and this rule alone:

```
"\n;"        → Error: The root node must have the first node.
"\n;\n\n"    → Error: The root node must have the first node.
"\n;a {}"    → ";a {}", no warning left
"\n"         → "\n", no warning
```

Without `fix` the file is reported as any other opening with a break. A configuration listing this rule beside `max-empty-lines` or `no-extra-semicolons` throws the same way in either order, since the error is raised inside the fix callback before any neighbour has a say.

## What would end it

Writing the break out of the raw that holds it — the root's trailing raw where the root has no node — or passing such a root over in the check, the way the whitespace-only one is passed over.
